modeshape-jcr/src/main/java/org/modeshape/jcr/index/local/ManagedLocalIndex.java

```diff
@@ -85,4 +93,17 @@ public void removeAll() {
         index.removeAll();
     }
 
+    @Override
+    public IndexManager.IndexStatus getStatus() {
+        return status.get();
+    }
+
+    @Override
+    public void updateStatus(IndexManager.IndexStatus currentStatus, IndexManager.IndexStatus newStatus) {
+        this.status.compareAndSet(currentStatus, newStatus);
```

Comment:
It would be fine to not check it if there were only 2 states, but with more states it's important to know if one of the unexpected state transitions happens. If we don't, then should that happen what state will it be left in?

What do you think about either making an assertion after the `compareAndSet` to ensure that it was set indeed set properly, or throw an exception if `compareAndSet` returned `false`?